### PR TITLE
Use ServiceMonitor config to deploy Prometheus in K8s

### PR DESCRIPTION
**Pull request by Deflaimun**
Fixes https://github.com/redpanda-data/documentation-private/issues/1651

Relies on https://github.com/redpanda-data/redpanda/issues/13321 being resolved.

### CODE REVIEW

File: modules/manage/partials/monitoring.adoc [hunk 1]
```diff
@@ -27,11 +27,21 @@ https://prometheus.io/[Prometheus^] is a system monitoring and alerting tool. It
 
 NOTE: Redpanda recommends creating monitoring dashboards with `/public_metrics`.
 
+:env-kubernetes:
+
 ifdef::env-kubernetes[]
 
-To configure Prometheus to monitor Redpanda metrics, you can use the https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus[Prometheus Helm chart^]:
+To configure Prometheus to monitor Redpanda metrics in Kubernetes, you can use the https://prometheus-operator.dev/[Prometheus Operator^]:
```
Comment (Deflaimun) [inline modules/manage/partials/monitoring.adoc:32]:
```suggestion
To configure Prometheus to monitor Redpanda metrics in Kubernetes, you can use the https://prometheus-operator.dev/[Prometheus Operator^]: 
```

**Comment by Feediver1** [comment]
@Deflaimun I noticed that the Netlify previews were using the URL for the updated files in the previews, but now is not. Does this need to be explicitly configured for public?

**Comment by Deflaimun** [comment]
@Feediver1 preview was missing a key config. This has been addressed.

File: modules/manage/partials/monitoring.adoc [hunk 2]
```diff
@@ -48,11 +58,9 @@ metadata:
...
   chartRef: {}
   clusterSpec:
-    statefulset:
-      annotations:
-        "prometheus.io/scrape": "true"
-        "prometheus.io/path": "public_metrics"
-        "prometheus.io/port": "<admin-api-port>"
+    monitoring:
+      enabled: true
+      scrapeInterval: 30s
```
Comment (alejandroEsc) [inline modules/manage/partials/monitoring.adoc:81]: I have a PR making this optional now, no longer required (scrapeInterVal). We should make a note that if you do not install prometheus, the servicemonitor will not be created even if you set enabled: true. This behavior may be changed soon however.

File: modules/manage/partials/monitoring.adoc [hunk 3]
```diff
@@ -99,24 +106,129 @@ helm upgrade --install redpanda redpanda/redpanda \
...
+
+. Wait until Prometheus is ready:
++
+[,bash]
+----
+kubectl get --namespace default prometheus prometheus --watch
 ----
 
-. Install Prometheus:
+. Deploy a NodePort Service for external access to Prometheus:
```
Comment (alejandroEsc) [inline modules/manage/partials/monitoring.adoc]: Managing prometheus and getting access to it externally seems like it belongs to pointers for prometheus specific documentation. Not sure we need to here, we probably can point instead to just using port forwarding with kubectl as most documentation for prometheus does.

**Review by alejandroEsc — APPROVED** [review]
lgtm, just a few comments

File: modules/manage/partials/monitoring.adoc (hunk 2 above)
Comment (alejandroEsc) [reply modules/manage/partials/monitoring.adoc:81]: OK my team agreed that we should never guard users like this. So we will explicitly fail if you enable monitoring but do not have the CRDs installed.

**Review by alejandroEsc — APPROVED** [review]
lgtm

**Review by Feediver1 — APPROVED** [review]
lgtm

**Pull request merged**